app/controllers/staff/budgets_controller.rb

```diff
@@ -28,8 +28,38 @@ def create
     end
   end
 
+  def edit
+    @budget = Budget.find(id)
+    authorize @budget
+
+    @activity = Activity.find(activity_id)
+  end
+
+  def update
+    @budget = Budget.find(id)
+    @activity = Activity.find(activity_id)
+    @budget.activity = @activity
+    authorize @budget
+
+    @budget.assign_attributes(budget_params)
+    @budget.value = monetary_value
```

Comment:
This has no bearing on your work, but I am wondering if we should do something in `monetary_value` to format the display? I work on the principle that the value is stored in the lowest possible unit, so pence for us and converted for display – I've not looked at what we do but the `to_f` in `monetary_value` feels a bit brutal.

Reply:
The `to_f` is brutal and I think it's actually redundant as the database decimal type will do a more precise conversion for us since it has `scale: 2`. I think it was intended to deal with the fact that users could provide more than 2 decimal places but this should probably be validated rather than chopped!

Reply:
Should we open a refactor card to address this?

Reply:
I'll add a card. I suspect it will go to the icebox to die but it's there if this does become an issue as a prompt.